java-squid/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -84,4 +90,107 @@ public boolean isSubtype(JavaType t, JavaType s) {
     return result;
   }
 
+  /**
+   * Compute the "Least Upper Bound" ("lub", jls8 §4.10.4) of a list of type. The "lub" is a shared supertype that is more specific than any
+   * other shared supertype (that is, no other shared supertype is a subtype of the least upper bound)
+   *
+   * FIXME Parameterized types are currently ignored, as the method is used only to handle Union Types Trees, themselves being used only
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)